sdk/core/System.ClientModel/api/System.ClientModel.net6.0.cs

```diff
@@ -188,7 +188,10 @@ public partial class ModelReaderWriterOptions
         public ModelReaderWriterOptions(string format) { }
         public string Format { get { throw null; } }
         public static System.ClientModel.Primitives.ModelReaderWriterOptions Json { get { throw null; } }
+        public object? ProxiedModel { get { throw null; } set { } }
```

Comment:
I understand now why this is settable -- it is so that a proxy model's implementation of Write can pass a model instance into the proxy's write routine.  This has to be settable to enable recursing to properties on the model that are different types.

I'm a little concerned that we're relying on the proxy model logic to "remember" to set this value, and writing internal shared source files to support that.  That will make it hard for end users to correctly write proxy types themselves, especially if they don't have access to our shared source files.  Is there a way to "bake in" the setting of this value into public types in SCM?